src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const initialState = state;
+
+  actions.forEach(object => {
+    if (object.type === 'addProperties') {
+      for (const key in object.extraData) {
+        initialState[key] = object.extraData[key];
+      }
+    } else if (object.type === 'removeProperties') {
+      for (const key of object.keysToRemove) {
+        delete initialState[key];
+      }
+    } else {
+      for (const key in initialState) {
+        delete initialState[key];
+      }
+    }
+  });
+
+  return initialState;
```

Comment:
Also this doesn't do much here as you are mutating the original object, it will either way be changed with the difference you return it from this function. If you did something like 
```
 const state = { x: 1 };

 transformState(state, [
   { type: 'addProperties', extraData: { yet: 'another property' } }
   { type: 'clear' },
   { type: 'addProperties', extraData: { foo: 'bar', name: 'Jim' } }
 ]);
``` 
The state object will be changed after the function call, in your case you are also piping it through the function.